apps/balancer-tools/src/app/poolsimulator/(types)/index.ts

```diff
@@ -0,0 +1,37 @@
+export interface TokensData {
+  symbol: string;
+  balance: number;
+  decimal: number;
+  rate?: number;
+  weight?: number;
+}
+
+export interface MetaStableParams {
+  ampFactor?: number;
+  swapFee?: number;
+}
+
+export interface GyroEParams {
```

Comment:
Don't these types already exist in the math package?

Reply:
You are correct. I wasn't sure how to transform the expected type from the subgraph(strings) to the ones we expect on the form(number or undefined). But now I use

```
type NumberGyroEParams<T> = {
  [K in keyof T]?: number;
};

export type GyroEParams = NumberGyroEParams<
  GyroEParamsFromSubgraph & DerivedGyroEParamsFromSubgraph
>;
```

To achieve the same as before. WDYT?

Reply:
This is looking good! Thanks for explaining it 😉